e/tool/tele/cli/helm.go

```diff
@@ -61,12 +61,19 @@ func (o *repoAddOptions) repoAdd() error {
 	lockCtx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
 	defer cancel()
 	locked, err := fileLock.TryLockContext(lockCtx, time.Second)
-	if err == nil && locked {
-		defer fileLock.Unlock()
-	}
 	if err != nil {
 		return trace.Wrap(err)
 	}
+	if !locked {
+		return trace.LimitExceeded("unable to obtain file lock")
+	}
+	defer func() {
+		errUnlock := fileLock.Unlock()
+		if err != nil {
+			return
+		}
+		err = errUnlock
+	}()
```

Comment:
It feels ugly that the unlock error is conditionally ignored.  Typically in this case we'd log, e.g:

https://github.com/gravitational/gravity/pull/2488/files#diff-1e407a6245149757c630df2b55ac3e9d17db91779569d981c4d77f5536b9a0afR127-R135

Why isn't that the approach here?

Reply:
The linked diff implements an error handler that logs the error - here, it makes sense to fail the function as a result. If you mean that the error unlocking should be logged in case another error is active - I'll update that.